hilti/runtime/src/types/bytes.cc

```diff
@@ -205,6 +205,24 @@ uint64_t Bytes::toUInt(ByteOrder byte_order) const {
     return i;
 }
 
+double Bytes::toReal() const {
+    // Ensure there are no null bytes inside our data, so that we can call strtod().
+    if ( Base::find('\0') != Base::npos )
+        throw InvalidValue("cannot parse real value: null byte in data");
+
+    const char* cstr = Base::c_str();
+    char* endp = nullptr;
+
+    errno = 0;
+    auto d = strtod(cstr, &endp);
```

Comment:
I was about to suggest `std::from_chars`, but float parsing is currently unsupported on libc++ 😞

Reply:
Yeah, it's a pain. :-(